shell/src/context_listener.rs

```diff
@@ -3,21 +3,31 @@
 
 //! Listens for events from the `protocol_runner`.
 
+use bytes::{Buf, BufMut, BytesMut};
```

Comment:
There are many clippy warnings regarding unused and duplicated imports on this file